src/others/observable/lv_observable.h

```diff
@@ -0,0 +1,263 @@
+    float fnum; /**< Floating point number */
+    const void* ptr; /**< Constant pointer  (string buffer, format string, font, cone text, etc)*/
+    lv_color_t color; /**< Color */
+} lv_subject_value_t;
+
+/**
+ * The subject (an observable value)
+ */
+typedef struct {
+    lv_ll_t subs_ll; /**< Subscribers*/
+    size_t type; /**< For elementary types the type id, for strings LV_SUBJECT_TYPE_STRING + the length of the string*/
+    lv_subject_value_t value; /**< Actual value*/
+} lv_subject_t;
+
+/**
+ * The observer object: a descriptor returned when subscribing LVGL widgets to subjects
+ */
+typedef struct {
+    lv_subject_t* subject; /**< The observed value */
+    lv_observer_cb_t cb; /**< Callback that should be called when the value changes*/
+    lv_obj_t* obj; /**< LVGL widget which has subscribed for the value*/
+    void* data1; /**< Additional parameter supplied when subscribing*/
+    void* data2; /**< Additional parameter supplied when subscribing*/
+} lv_observer_t;
+
+typedef uint32_t lv_bitmask_pattern_t;
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Initialize an int type subject
+ * @param subject address of the subject
+ * @param value initial value
+ */
+void lv_subject_init_int(lv_subject_t* subject, int32_t value);
+
+/**
+ * Initialize a pointer type subject
+ * @param subject address of the subject
+ * @param value initial value
+ */
+void lv_subject_init_ptr(lv_subject_t* subject, void* value);
+
+/**
+ * Initialize a string type subject
+ * @param subject address of the subject
+ * @param buf pointer to a buffer containing the string value
+ * @param size size of the buffer
+ */
+void lv_subject_init_string(lv_subject_t* subject, char* buf, size_t size);
+
+/**
+ * Initialize a color type subject
+ * @param subject address of the subject
+ * @param color initial value
+ */
+void lv_subject_init_color(lv_subject_t* subject, lv_color_t color);
+
+/**
+ * Subscribe an LVGL widget to a subject
+ * @param subject address of the subject
+ * @param cb callback to call when the value of the subject changes
+ * @param obj the LVGL object what will be affected by the change
+ * @param data1 optional additional parameter stored in the observer object
+ * @param data2 optional additional parameter stored in the observer object
+ * @return the observer object
+ * @note This is a generic interface, which is normally not called directly. For specific purposes there are specific wrappers.
+ * @note The observer object is allocated on the heap. It will be freed automatically when the object is deleted or when it is unsubscibed from the subject.
+ */
+void* lv_subject_subscribe_obj(lv_subject_t* subject, lv_observer_cb_t cb, lv_obj_t* obj, void* data1, void* data2);
+
+/**
+ * Remove observer from the list of subscribers
+ * @param subject address of the observer object returned when a widget was subscribed to the subject
```

Comment:
```suggestion
 * @param observer address of the observer object returned when a widget was subscribed to the subject
```

Reply:
Good catch. Actually, the implementation was using a wrong name as well.